src/traces/barpolar/attributes.js

```diff
@@ -0,0 +1,54 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var scatterPolarAttrs = require('../scatterpolar/attributes');
+var barAttrs = require('../bar/attributes');
+
+module.exports = {
+    r: scatterPolarAttrs.r,
+    theta: scatterPolarAttrs.theta,
+    r0: scatterPolarAttrs.r0,
+    dr: scatterPolarAttrs.dr,
+    theta0: scatterPolarAttrs.theta0,
+    dtheta: scatterPolarAttrs.dtheta,
+    thetaunit: scatterPolarAttrs.thetaunit,
+
+    // orientation: {
+    //     valType: 'enumerated',
+    //     role: 'info',
+    //     values: ['radial', 'angular'],
+    //     editType: 'calc+clearAxisTypes',
+    //     description: 'Sets the orientation of the bars.'
+    // },
+
+    base: barAttrs.base,
+    offset: barAttrs.offset,
+    width: barAttrs.width,
+
+    text: barAttrs.text,
+    // hovertext: barAttrs.hovertext,
```

Comment:
I guess `hovertext` is omitted just because, without on-graph text it's equivalent to `text`? That's fine, but we probably want to give `text` a separate description from the one it's inheriting from `scatter`. I haven't looked at all the other descriptions to see if they make sense as inherited...

If/when we do implement on-graph text, we can probably reuse logic from `pie` to fit text into annular sectors (probably not the outside text logic though).

Reply:
> I guess hovertext is omitted just because, without on-graph text it's equivalent to text? 

Exactly. 

> That's fine, but we probably want to give text a separate description from the one it's inheriting from scatter

Good call 

> I haven't looked at all the other descriptions to see if they make sense as inherited...

`barpolar` width, base and offset aren't exactly equivalent to the ones for `bar`

Improvements in https://github.com/plotly/plotly.js/pull/2954/commits/01e15b401f3e22d930848822fbce9b62a9ff499b